cmake/list_append_keys_from.cmake

```diff
@@ -0,0 +1,22 @@
+#
+# Append key[=value] elements from one list to another list if they are not already in the target list.
```

Comment:
I just copied it from list_append_unique and missed adding proper argument document, I can add this after we have decided the parameter list below.